src/backend/gl/src/lib.rs

```diff
@@ -656,14 +656,31 @@ impl CommandQueue {
             },
             Command::CopyBuffer(src, dst, src_offset, dst_offset, size) => {
                 let gl = &self.share.context;
-                unsafe {
-                    gl.BindBuffer(gl::COPY_READ_BUFFER, src);
-                    gl.BindBuffer(gl::COPY_WRITE_BUFFER, dst);
-                    gl.CopyBufferSubData(gl::COPY_READ_BUFFER,
-                                         gl::COPY_WRITE_BUFFER,
-                                         src_offset,
-                                         dst_offset,
-                                         size);
+
+                if self.share.capabilities.copy_buffer_supported {
+                    unsafe {
+                        gl.BindBuffer(gl::COPY_READ_BUFFER, src);
+                        gl.BindBuffer(gl::COPY_WRITE_BUFFER, dst);
+                        gl.CopyBufferSubData(gl::COPY_READ_BUFFER,
+                                            gl::COPY_WRITE_BUFFER,
+                                            src_offset,
+                                            dst_offset,
+                                            size);
+                    }
+                } else {
+                    assert!(size > 0);
+
+                    unsafe {
+                        let mut src_ptr = 0 as *mut ::std::os::raw::c_void;
+                        factory::temporary_ensure_mapped(&mut src_ptr, gl::COPY_READ_BUFFER, src, memory::READ, gl);
```

Comment:
You are bypassing GFX's mapping tracking mecanism so the [automatic unmapping](https://github.com/gfx-rs/gfx/blob/master/src/backend/gl/src/lib.rs#L877) will not happen and the buffer might already be mapped. IIRC this means that the next time you will try to use this buffer for something like mapping it again or drawing with it, you will get an `INVALID_OPERATION`.